src/rules/typeColonSpacing/evaluateVariables.js

```diff
@@ -0,0 +1,31 @@
+import _ from 'lodash';
+import {getParameterName, quoteName} from '../../utilities';
+
+export default (context, report) => {
+  const sourceCode = context.getSourceCode();
+
+  const getColon = (node, typeAnnotation) => {
+    if (node.type === 'FunctionTypeParam') {
+      return sourceCode.getFirstToken(node, node.optional ? 2 : 1);
+    } else {
+      return sourceCode.getFirstToken(typeAnnotation);
+    }
+  };
```

Comment:
Is this function left over from somewhere else? Just `sourceCode.getFirstToken(typeAnnotation)` works.